contracts/swap/swap.go

```diff
@@ -105,29 +101,43 @@ func InstanceAt(address common.Address, backend Backend) (Contract, error) {
 }
 
 // Withdraw withdraws amount from the chequebook and blocks until the transaction is mined
-func (s simpleContract) Withdraw(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
+func (s simpleContract) Withdraw(auth *bind.TransactOpts, amount *big.Int) (*types.Receipt, error) {
 	tx, err := s.instance.Withdraw(auth, amount)
 	if err != nil {
 		return nil, err
 	}
-	return WaitFunc(auth, backend, tx)
+	return WaitFunc(auth, s.backend, tx)
 }
 
-// Deposit sends a transaction to the chequebook, which deposits the amount set in Auth.Value and blocks until the transaction is mined
-func (s simpleContract) Deposit(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
-	rawSimpleSwap := contract.SimpleSwapRaw{Contract: s.instance}
-	if auth.Value != big.NewInt(0) {
-		return nil, fmt.Errorf("Deposit value can only be set via amount parameter")
-	}
-	if amount == big.NewInt(0) {
+// Deposit sends an amount in ERC20 token to the chequebook and blocks until the transaction is mined
+func (s simpleContract) Deposit(auth *bind.TransactOpts, amount *big.Int) (*types.Receipt, error) {
+	var zero big.Int
```

Comment:
Good suggestion. Thanks :+1:

Reply:
watch out: this `var zero` business was occurring in two places, and you removed it from both.

this is causing a compilation failure on `swap.go:224`.

Reply:
Hmm. I see. I saw this as well already and decided to just put everywhere &big.int{} (also at the places where we re-used var zero). I think it is cleaner to read. Tell me if you have objections :)

Reply:
I personally don't think it's easier to read, actually the opposite

Reply:
It is not a dealbreaker for me, but my reasons for prefering &big.int{} directly:
- We might remove at one place the usage of the variable and then remove the variable declaration because we think it is not needed anymore (happened to me)
- We might have the idea the &zero is something build in into Golang for the zero-values of variables (happened to me)
- It is not very nice to see a variable being used at one place and then look up where this variable is declared and what it is.
- In terms of efficiency, there is no difference, because we only use the variable once after declaration (and if we would re-use, there is probably only a marginal difference).

Reply:
Efficiency is indeed not an issue here, but I don't think any of the above criteria really have much weight.

But there is one criteria which makes the `^big.Int{}` notation indeed better: `zero` is used like a constant but it is none, someone could even assign it a value (actually unlikely because of the name but possible)  and we could still use it further down thinking it is the empty value.